gnomad/utils/filtering.py

```diff
@@ -533,40 +533,47 @@ def split_vds_by_strata(
     }
 
 
-def filter_freq_by_meta(
-    freq_expr: hl.expr.ArrayExpression,
-    freq_meta_expr: hl.expr.ArrayExpression,
+def filter_arrays_by_meta(
+    meta_expr: hl.expr.ArrayExpression,
+    meta_indexed_exprs: Dict[str, hl.expr.ArrayExpression],
     items_to_filter: Union[Dict[str, List[str]], List[str]],
     keep: bool = True,
     combine_operator: str = "and",
-) -> Tuple[hl.expr.ArrayExpression, hl.expr.ArrayExpression]:
+) -> Tuple[hl.expr.ArrayExpression, Dict[str, hl.expr.ArrayExpression]]:
     """
-    Filter frequency and frequency meta expressions specified by `items_to_filter`.
+    Filter both metadata array expression and meta data indexed expression by `items_to_filter`.
 
     The `items_to_filter` can be used to filter in the following ways based on
-    `freq_meta_expr` items:
+    `meta_expr` items:
     - By a list of keys, e.g. ["sex", "downsampling"].
     - By specific key: value pairs, e.g. to filter where 'pop' is 'han' or 'papuan'
     {"pop": ["han", "papuan"]}, or where 'pop' is 'afr' and/or 'sex' is 'XX'
     {"pop": ["afr"], "sex": ["XX"]}.
 
-    The items can be kept or removed from `freq_expr` and `freq_meta_expr` based on the
-    value of `keep`.
+    The items can be kept or removed from `meta_indexed_expr` and `meta_expr` based on
+    the value of `keep`. For example if `meta_indexed_exprs` is {'freq': ht.freq,
+    'freq_meta_sample_count`: ht.index_globals().freq_meta_sample_count} and `meta_expr`
+    is ht.freq_meta then if `keep` is True, the items specified by `items_to_filter`
+    such as  'pop' = 'han' will be kept and all other items will be removed from the
+    ht.freq, ht.freq_meta_sample_count, and ht.freq_meta.
 
```

Comment:
I was testing your code, I have two questions about sample_count:
1. If it's indexed by freq_meta, should it have the same length as freq_meta? I found different length before filtering.  
2. was it called `freq_sample_count` in v3, now we change it to `freq_meta_sample_count`?

Reply:
1. It should have the same length, what HT are you testing on?
2. Yes, its more clear in my opinion.

Reply:
I tested 2 HT: 
v3 release HT and an new HT I generated for the added HGDP/TGP: 
```commandline
from gnomad_qc.v3.resources.release import release_ht_path
ht = hl.read_table(release_ht_path(release_version="3.1.2"))
```
and 
```
ht = hl.read_table(
    'gs://gnomad-tmp/gnomad.exomes.v4.0.qc_data/gnomad.genomes.v4.0.hgdp_1kg_subset_updated_callstats_added.ht')
```
they both didn't have equal length before filtering.

Reply:
I think you found a bug in v3.1.2. I'm fairly certain we'd want freq_sample_count to match freq_meta unless there is some sort of implied understanding of sample counts in downsamplings and those arent included? maybe @jkgoodrich knows why it doesnt? If 3.1.2 is incorrect and you used in for your subset, then it will also be incorrect right?

Reply:
My guess is that the freq_sample_counts do not include the hgdp tgp freq groups

Reply:
it doesn't add up though, 598 - hgdp (145)  - tgp (82) = 371 (!= 201)

Reply:
Maybe also - subsets will get there

Reply:
Just to follow this up with code showing we didn't concatenate the sample count global for the subsets, only the freq_meta
 https://github.com/broadinstitute/gnomad_qc/blob/0c52cf47e48fa5b503d874e96482ea4286474c71/gnomad_qc/v3/create_release/create_release_sites_ht.py#L293

Reply:
Qin, can you add a ticket to update this for the v4 genomes?

Reply:
here: https://app.zenhub.com/workspaces/gnomad-5f4d127ea61afc001d6be50b/issues/gh/broadinstitute/gnomad_production/1117